service/src/main/java/bio/terra/workspace/common/utils/WorkspaceActivityLogHook.java

```diff
@@ -67,6 +67,11 @@ public HookAction endFlight(FlightContext context) {
         context
             .getInputParameters()
             .get(WorkspaceFlightMapKeys.OPERATION_TYPE, OperationType.class);
+    if (operationType == null) {
+      // If a flight is directly launched from stairway and skipped JobService, operation type could
+      // still be null.
+      return HookAction.FAULT;
```

Comment:
Context is, I saw this in [test logs](https://scans.gradle.com/s/7katupx2qemc4) and was confused that flight succeeded:

```
    2022-06-28 16:19:28.852   WARN 65708 --- [ool-7-thread-10] bio.terra.stairway.impl.HookWrapper      : Stairway Hook failed with exception

    bio.terra.workspace.service.workspace.exceptions.MissingRequiredFieldsException: Missing required field
        at bio.terra.workspace.db.model.DbWorkspaceActivityLog.lambda$static$0(DbWorkspaceActivityLog.java:13)
        at java.base/java.util.Optional.orElseThrow(Optional.java:403)
        at bio.terra.workspace.db.model.DbWorkspaceActivityLog.getOperationType(DbWorkspaceActivityLog.java:21)
        at bio.terra.workspace.db.WorkspaceActivityLogDao.writeActivity(WorkspaceActivityLogDao.java:40)
```

We're already logging the exception, how about just some comments at the top to address Melissa confusion?

/**
 * Note: If flight is started from stairway directly (not through JobService) and an exception is thrown (such as MissingRequiredFieldsException for missing operationType), the flight will still succeed. Currently, there is no way for the hook to fail the flight; returning HookAction.FAULT doesn't do anything.
 */

Reply:
actually... i think that not having operationType here might be OK and we don't need to handle it (return early). There are some sub-flight cases for cloning where we start a flight from stairway  inside a clone flight. 

1. if flight starts from jobservice, operationtype is always guarantee to be not null by `jobbuilder`
2. if flight starts from stairway, operationtype can be null. Currently this only happens in test and subflight. We can just ignore them in those two cases? 

wdyt?

Reply:
That is OK with me. Maybe we need a JobService endpoint for launching sub-flights instead of doing it directly against Stairway. Then we could have consistent validity checking.